packages/core/src/app/ovm/deciders/signed-by-decider.ts

```diff
@@ -0,0 +1,137 @@
+import { Decision } from '../../../types/ovm'
+import { KeyValueStoreDecider } from './key-value-store-decider'
+import { DB } from '../../../types/db'
+import { CannotDecideError } from './utils'
+
+export interface SignedByInput {
+  publicKey: Buffer
+  message: Buffer
+}
+
+export interface SignedByWitness {
+  signature: Buffer
+}
+
+export type SignatureVerifier = (
+  publicKey: Buffer,
+  message: Buffer,
+  signature: Buffer
+) => Promise<boolean>
+
+/**
+ * Decider that determines whether the provided witness is the provided message signed by
+ * the private key associated with the provided public key.
+ */
+export class SignedByDecider extends KeyValueStoreDecider {
+  private static readonly UNIQUE_ID = 'SignedByDecider'
+
+  private readonly signatureVerifier: SignatureVerifier
+
+  constructor(db: DB, signatureVerifier: SignatureVerifier) {
+    super(db)
+
+    this.signatureVerifier = signatureVerifier
+  }
+
+  public async makeDecision(
+    input: SignedByInput,
+    witness: SignedByWitness
+  ): Promise<Decision> {
+    if (!witness) {
+      throw new CannotDecideError(
+        'Cannot decide whether message was signed without witness.'
+      )
+    }
+
+    const signatureMatches: boolean = await this.signatureVerifier(
+      input.publicKey,
+      input.message,
+      witness.signature
+    )
+
+    if (signatureMatches) {
+      await this.storeDecision(
+        input,
+        SignedByDecider.serializeDecision(witness, input, signatureMatches)
+      )
+    }
+
+    return this.constructDecision(
+      witness.signature,
+      input.publicKey,
+      input.message,
+      signatureMatches
```

Comment:
It is not quite the case that, if the signature doesn't match, that we can decide "false"--we can only fail to make the decision.  Intuitively, "just because this signature isn't right doesn't mean the right one isn't out there."

Reply:
Ahh ok perhaps I misunderstood the motivation for that decider -- I had assumed that it was to determine if X was signed by Y, not whether or not there exists some Y that has signed X. Looking at the input vs witness, I see that your point is totally valid.  This is pulled in from the other PR ( #378 ) since it's a dependency and that one isn't approved, so I'll make sure to add it there for reference.